### ISSUE TITLE
Catkin Tools

### ISSUE THREAD

**Issue by oxoocoffee** [ros-industrial/ros_qtc_plugin#103]
I was wondering if catkin tools is supported in current version.
After building your plug-in with Qt 5.8 I wanted to experiment. So I created simple workspace using you plugin. I selected catkim tools instead of carkin make and pressed finish. Then I did try to add package. Looks like package was created (I can see it in terminal) but in Qt Creator workspace it was not added. 
Is this know issue? I did install python-catkin-tools before starting Qt Creator.

**Comment by oxoocoffee** [comment]
I just realized that initial workspace creation did not create src folder. So when I added package it did not put it to src by default.

**Comment by Levi-Armstrong** [comment]
When you added the package it should have a place where you specify the location. Using the tool it would let you create the src directory if it did not exist. So right now it does not do any checking, it is left up to the user to make sure they select the correct location. I should probably add a check to make sure at least it resides in the src directory.

**Comment by oxoocoffee** [comment]
I think since this is "preferred" way of doing it I think creating src folder when creating workspace would make sense. Going further when creating package this wizard would automatically by default point to src where package would be created. I also think adding msg, srv, action .... folders when creating package would be great. What do you think? If you are busy I could do this and do pull request.

**Comment by Levi-Armstrong** [comment]
It looks like using catkin_make the src directory gets created, but when using catkin tools it does not, which is a bug.

I agree with that when a user adds a package the path should defaults to the workspace src directory and this should not be to difficult to implement.

When creating a package under the hood I am executing the command catkin_create_pkg which may do this assuming you pass the right depends during creation but I have not tested this. Though, I have been contemplating moving away from using this to provide more advanced features as you suggested. If you are interested on working on this I can provide support.

**Comment by oxoocoffee** [comment]
Ok I will look into this. Do you have a script that just builds your plugin and not entire Qt Creator? Or what needs to be changed. I can modify your script to optionally download Qt Creator if you need to build. What about project itself?

**Comment by Levi-Armstrong** [comment]
>Ok I will look into this. Do you have a script that just builds your plugin and not entire Qt Creator? Or what needs to be changed. I can modify your script to optionally download Qt Creator if you need to build. What about project itself?

I looked at the setup script and I think the only thing that needs to be updated is this [line](https://github.com/ros-industrial/ros_qtc_plugin/blob/master/setup.sh#L35) which should now have 4.2 instead of 4.1. Then if a developer follows the wiki for developers it should work with this change but I have not tested it.

**Comment by Levi-Armstrong** [comment]
@oxoocoffee You ok with closing this issue?

**Comment by oxoocoffee** [comment]
I ma trying to add missing folders sometime today unless you already did this. If you did this I can close it else I will do pull request when I am done.

**Comment by Levi-Armstrong** [comment]
I just created a PR that adds the src folder for catkin tools, but I would recommend closing this one and opening a new issue with a new description for adding additional folders if you don't mind.

**Comment by oxoocoffee** [comment]
No problem
